deepcell/applications/application.py

```diff
@@ -277,35 +308,31 @@ def _predict_segmentation(self,
             ValueError: Input data must match required rank of the application, calculated as
                 one dimension more (batch dimension) than expected by the model
 
+            ValueError: Input data must match required number of channels of application
+
         Returns:
             np.array: Labeled image
         """
 
         # Check input size of image
         if len(image.shape) != self.required_rank:
-            raise ValueError('Input data must have {} dimensions'
+            raise ValueError('Input data must have {} dimensions. '
                              'Input data only has {} dimensions'.format(
                                  self.required_rank, len(image.shape)))
 
-        # Resize image, returns unmodified if appropriate
-        image, original_shape = self._resize_input(image, image_mpp)
```

Comment:
I don't like having the first resize hidden inside `_run_model` while the second is exposed in `_predict_segmentation`. I would vote for keeping both resize steps in `_predict_segmentation`.